src/main/java/com/cburch/logisim/LogisimVersion.java

```diff
@@ -94,8 +94,7 @@ private LogisimVersion initFromVersionString(String versionString) throws Illega
           }
           suffix = sufStr;
         } else if (sufStr.length() > 1) {
-          // for single character suffix it can only be letter
-          pattern = "^(-)?([a-z]+([a-z\\d])*)+$";
+          pattern = "^(-)?([a-z][a-z\\d]*)$";
```

Comment:
Now that I've slept on it, I wake up wondering if you start with "^(-?)", will it be necessary to check whether the sep is null? My guess is that it will simply be the empty string then.

I also wonder if the first pattern above can be used to match everything and not have to do two followup parsing of parts. I'd suggest there something like:
`^(\\d*)\\.(\\d*)\\.(\\d*)(-?)([a-z][a-z\\d]*)$`. Then you would have 5 parts.

Note that I am guessing about the way to insist on the dots (which you don't do above since a dot seems to match any character). A google search shows that should work. `[.]` is another option.

I think this method can be simplified a fair bit.

Reply:
Hmmm. I missed the optional [a-z]... Try `^(\\d*)\\.(\\d*)\\.(\\d*)(-?)(([a-z][a-z\\d]*)?)$`.

Reply:
I do agree, it's not optimal code, but I refuse to work on this any more. If you have fun optimizing, then please send separate PR